hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/viewfs/ViewFs.java

```diff
@@ -919,6 +921,42 @@ public FSDataOutputStream createInternal(final Path f,
         FileAlreadyExistsException, FileNotFoundException,
         ParentNotDirectoryException, UnsupportedFileSystemException,
         UnresolvedLinkException, IOException {
+      Preconditions.checkNotNull(f, "File cannot be null.");
+      // Just a sanity check. This should not happen.
```

Comment:
What do you mean by : This should not happen. Ideally client call shouldn't land here or This If check will never get hit(It does get hit, I verified),

Reply:
My comments seems to be too light. "//" will trigger this. But ideally that should not happen as users would not create files with /. However if somebody attempts we are ready to shoot. I have removed that comment, as it may be confusing.
However, this will trigger mostly in ViewFs.java, but may not be in ViewFileSystem.java. In ViewFileSystem.java, getDefaultReplication API itself might shade this condition if parent is internal currently. Anyway the current condition should be good enough to handle if some one try with files as "//" etc
Thanks for checking.